spec/models/event_spec.rb

```diff
@@ -40,4 +40,30 @@
     end
 
   end
+
+  describe '#in_week_group_by_weekday' do
+    let(:start_time) { Time.now.beginning_of_week }
+    let!(:event1) { create(:event, start_time: start_time, end_time: start_time + 2.hours) }
+    let!(:event2) { create(:event, start_time: start_time + 1.days, end_time: start_time + 2.days)}
+    let!(:event3) { create(:event, start_time: start_time + 3.days, end_time: start_time + 4.days)}
+
+    let(:expected_events) do
+      {
+        0 => [event1],
+        1 => [event2]
+      }
+    end
+
+    let(:not_expected_events) {
+      {
+        9 => [event3]
+      }
+    }
+
+    it "returns all events from specified week grouped by weekday" do
+      events = described_class.in_week_group_by_weekday(start_time)
```

Comment:
I'd replace this: 
`events = described_class.in_week_group_by_weekday(start_time)`
with 
`subject { described_class.in_week_group_by_weekday(start_time) }`